src/shogun/mathematics/graph/ops/Input.cpp

```diff
@@ -0,0 +1,9 @@
+#include <shogun/mathematics/graph/ops/Input.h>
+
+using namespace shogun;
+
+// so this is a thing
+// template<> template<> void InputImpl<ShogunInput>::evaluate(const SGVector<float32_t>&);
```

Comment:
do we need these, i mean since the header has everything...

Reply:
Not anymore, I figured tensor should be the type erase, any shape, representation.